### PR TITLE
Sentry: Sample Next.js client errors

### PR DESCRIPTION
**Pull request by masonmcelvain**
[Motivation in slack](https://ifixit.slack.com/archives/C04C05FA3EH/p1696266414063559)

[Docs](https://vercel.com/docs/projects/environment-variables/system-environment-variables#system-environment-variables) on system environment variables

## QA

Make sure errors are not reported to sentry by default in dev

CC @jarstelfox @sctice-ifixit

### CODE REVIEW

File: frontend/sentry.client.config.ts [hunk 1]
```diff
@@ -24,7 +25,7 @@ Sentry.init({
    },
    dsn: SENTRY_DSN,
    integrations: [new BrowserTracing()],
-   sampleRate: 1.0,
+   sampleRate: VERCEL_ENV === 'production' ? 0.05 : 1.0,
```
Comment (jarstelfox) [inline frontend/sentry.client.config.ts:28]: Should we only report in dev?

That's what we currently do in the mono repo, but I am unsure if there is enough of a reason to keep reporting in non-prod envs


```suggestion
   sampleRate: VERCEL_ENV === 'production' ? 0.05 : 0,
```

**Review by jarstelfox — DISMISSED** [review]
CR 🌵

File: frontend/sentry.client.config.ts (hunk 1 above)
Comment (masonmcelvain) [reply frontend/sentry.client.config.ts:28]: It's convenient to have error reporting turned on by default for QA and debugging/iterating. However, if we're worried about chipping away at our Sentry budget, I don't think QA would mind opting into error reporting like we do on the PHP side as long as we notified everyone about it.

Comment (jarstelfox) [reply frontend/sentry.client.config.ts:28]: I think it's weird that local dev errors make it to Sentry and even cause Sentry's new PR feature to misreport errors:

Example:
https://github.com/iFixit/react-commerce/pull/1976#issuecomment-1730229461


https://ifixit.sentry.io/issues/4494886521/?project=6475315&referrer=github-pr-bot

Background:
https://github.com/getsentry/sentry/discussions/49996#discussioncomment-6807526

Comment (masonmcelvain) [reply frontend/sentry.client.config.ts:28]: I [see](https://ifixit.sentry.io/settings/billing/overview/) our quota is 50k, and 4k is a significant chunk of that. I think turning off reporting in dev by default makes sense 👍🏻

Comment (jarstelfox) [reply frontend/sentry.client.config.ts:28]: See: https://ifixit.sentry.io/dashboard/65897/?project=6475315&statsPeriod=30d

I'd say more than half of all errors are from local dev.

The query there is looking at URL like `localhost`

Comment (danielbeardsley) [reply frontend/sentry.client.config.ts:28]: I agree. I almost *never* want to see dev errors and I practically only look at sentry to see prod errors. Seeing dev errors makes me generally less trusting of the stuff in sentry cause I'm constantly questioning... oh wait... did these happen in dev?

Comment (danielbeardsley) [reply frontend/sentry.client.config.ts:28]: Maybe we should take in a `SAMPLE_RATE variable and have it default to `0`? Then in vercel env conf we can bump it for just the `main` branch deployments.

Comment (jarstelfox) [reply frontend/sentry.client.config.ts:28]: Sounds like a great idea to me

File: frontend/sentry.client.config.ts [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
-// The config you add here will be used whenever a page is visited.
-// https://docs.sentry.io/platforms/javascript/guides/nextjs/
-
+import { VERCEL_ENV } from '@config/env';
 import { isCurrentProductionDeployment } from '@helpers/vercel-helpers';
 import * as Sentry from '@sentry/nextjs';
 import { BrowserTracing } from '@sentry/browser';
 
 const SENTRY_DSN = process.env.NEXT_PUBLIC_SENTRY_DSN;
+const sampleRate = VERCEL_ENV === 'production' ? 0.05 : 0;
```
Comment (danielbeardsley) [inline frontend/sentry.client.config.ts:7]: As far as I remember, *all* our vercel deployments are "production", so this includes all pull request code.

Comment (masonmcelvain) [reply frontend/sentry.client.config.ts:7]: The NODE_ENV is production, but not the VERCEL_ENV

image

**Comment by danielbeardsley** [comment]
CR :+1:

**Review by jarstelfox — APPROVED** [review]

**Comment by masonmcelvain** [comment]
QA ✌🏻 On main, I can produce an [error in sentry](https://ifixit.sentry.io/share/issue/de09d6e4572744af9b934ed2d0cfa74c/) in local dev. On this branch, I can't

**Pull request merged**